service/config_watcher.go

```diff
@@ -62,7 +68,6 @@ func (cm *configWatcher) onChange(event *configmapprovider.ChangeEvent) {
 	}
 }
 
-func (cm *configWatcher) close(ctx context.Context) error {
+func (cm *configWatcher) close() {
 	close(cm.watcher)
-	return cm.ret.Close(ctx)
```

Comment:
Why is this deleted?

Reply:
i have modified the code flow in a way that the config watcher is only created once in the lifecycle of the collector as compared to how it was originally implemented where a config watcher is created per change in the config file. once that change was made, it didnt make sense to close the retrieved and pass the error down. i moved that logic into a `get()` method that follows the `Retrieve()` -> `watch` -> `Close()` lifecycle.

Reply:
>i have modified the code flow in a way that the config watcher is only created once in the lifecycle of the collector

The new rearranged code is harder to follow and understand. Please refactor it to clearly show that the code follows lifecycle described in the `Provider` [comments](https://github.com/open-telemetry/opentelemetry-collector/blob/96a882a87742e2a921ebabf8e43ad97d8bf0e8c0/config/configmapprovider/provider.go#L55):
```
// The typical usage is the following:
//
//		r := mapProvider.Retrieve()
//		r.Get()
//		// wait for onChange() to be called.
//		r.Close()
//		r = mapProvider.Retrieve()
//		r.Get()
//		// wait for onChange() to be called.
//		r.Close()
//		// repeat Retrieve/Get/wait/Close cycle until it is time to shut down the Collector process.
//		// ...
//		mapProvider.Shutdown()
```
It was more visible before this change, admittedly it was not ideal but was better than what we have now. Now it is even harder to see that we are actually following the required lifecycle. All the current loop in runAndWaitForShutdownEvent shows is a watch, followed by `get()`.